### DISCUSSION TITLE
How to recover from "RTLSDR: cannot open device." on RasPi

### DISCUSSION THREAD

**Discussion by SFSailor** [jvde-github/AIS-catcher#339] [Q&A] [answered]
From time to time, particularly after an error, I get the message "RTLSDR: cannot open device."

Is there a way to clear this problem short of rebooting?
RasPi 5 running latest Debian.

**Comment by jvde-github** [comment]
Bit difficult to say without understanding the underlying reason. Is there another program or process competing for the device or perhaps it is a hardware issue with the stick?

If you run from a system daemon then typically the program will try to start until it is succesful. I have not seen or heard this issue before (also running a 5 myself)

**Reply by jvde-github** [reply]
is there btw  more  output that indicate the reason for the error? typically the rtlsdr library prints some messages

**Comment by SFSailor** [comment]
Yes, I am running a python script that receives the output via udp, lightly processes it, and forwards to a remote server.
I'll look for an error message. Unfortunately, it seems to be the python that throws the error and the message gets lost in the output from the ais-catcher.  Maybe I should suppress the output to the screen...

**Reply by jvde-github** [reply]
I see a potential issue if there is an issue with the device and the program stops and does not properly close the device resulting in issues with a restart. I will try to see if I can replicate. But if you manage to find any clues it would be helpful.

**Reply by jvde-github** [reply]
I just added a small fix that does an extra check that the device is always closed at termination.

**Comment by SFSailor** [comment]
Cool! thanks!

**Reply by jvde-github** [reply]
I did some stress testing trying to crash the program at several locations and then see if a next run cannot open the dongle. I could not replicate the issue unfortunately.

**Reply by SFSailor** [reply, marked as answer]
I think the crashing is happening on the python script I am running in parallel to post-process the data.  Guess I will get at it from the Linux side...
